docs/develop/secrets.md

```diff
@@ -1,9 +1,123 @@
 # Secret management
 
-{% hint style="warning" %}
+Some models need access to APIs, databases, AWS resources, or other secured information. With Truss, you can securely reference access keys, API tokens, passwords, secrets, and more. Keep this sensitive information out of your Git repository and use the same best practices for ML models as production websites.
+
+### Accessing secrets in models
+
+This example is based on the [GFP-GAN example Truss](https://github.com/basetenlabs/truss/tree/main/examples/gfpgan) in the main repository. That model needs access to an S3 bucket on AWS, requiring an access key id, access key secret, AWS region, and bucket name. These values are handled as secrets.
+
+{% hint style="danger" %}
+Never commit secret values to source control.
+{% endhint %}
+
+Start by defining a list of secrets in the `config.yaml` file. YAML is a key-value store, but you don't want to **ever** actually store the secret values in this file, even when running the Truss locally. Instead, set default values for the secrets, like `null`, and can leave comments about what kind of value to expect. Default values can also be fallback values that show format or type, like an empty string, the number 0, or an example UUID.
+
+```yaml
+secrets:
+  gfpgan_aws_access_key_id: null
+  gfpgan_aws_secret_access_key: null
+  gfpgan_aws_region: null # e.g. us-east-1
+  gfpgan_aws_bucket: null
+```
+
+Then, you can access the secrets in the `model/model.py` file by referencing them as kwargs in the init function.
+
+```python
+def __init__(self, **kwargs) -> None:
+    self._config = kwargs.get("config")
+    secrets = kwargs.get("secrets")
+```
+
+From there, you can use the secrets as a dictionary within the model file in any function.
+
+```python
+# Still in __init__
+self.s3_config = (
+    {
+        "aws_access_key_id": secrets["gfpgan_aws_access_key_id"],
+        "aws_secret_access_key": secrets["gfpgan_aws_secret_access_key"],
+        "aws_region": secrets["gfpgan_aws_region"],
+    }
+)
+self.s3_bucket = (secrets["gfpgan_aws_bucket"])
+```
+
+## Setting secrets locally
+
+When running your model locally there are two ways to set secret values.
+
+### Environment variables
+
+You can always set an environment variable and reference its value in your Truss. To avoid namespace conflicts, environment variable names must be prefixed with `TRUSS_SECRET_`. For example:
```

Comment:
Sorry, I remembered this incorrectly; environment variables based secrets binding only works in the docker image and baseten right now, not local.

Reply:
@pankajroark i'm curious, what would it take to make them work in local too? it could make it a more uniform experience for users if they don't have to remember this exception.